autoprotocol/protocol.py

```diff
@@ -126,11 +128,18 @@ class Protocol(object):
             }
     """
 
-    def __init__(self, refs=None, instructions=None, propagate_properties=False):
+    def __init__(
+        self,
+        refs=None,
+        instructions=None,
+        propagate_properties=False,
+        time_constraints=None,
+    ):
         super(Protocol, self).__init__()
         self.refs = refs or {}
         self.instructions = instructions or []
         self.propagate_properties = propagate_properties
+        self.time_constraints = time_constraints or []
```

Comment:
sorry going back on this, I think it should default to None, see below

Reply:
@yangchoo Actually, if we set this to None instead of [], either we'll need to add extra checks in LN#608, 615 and 624, or re-set the attribute from None to [] so we can have +=[...] operation. 
Kept it to [] for now, but can change it to None if we really want that...

Reply:
right.. I see. Okay then. Seems acceptable, at the cost of special casing required downstream